index.bs

```diff
@@ -2725,6 +2769,284 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
 
 1. Return [=success=] with data null.
 
+</div>
+#### The browsingContext.locateNodes Command ####  {#command-browsingContext-locateNodes}
+
+The <dfn export for=commands>browsingContext.locateNodes</dfn> command returns a
+list of all nodes matching the specified locators.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browsingContext.LocateNodes = (
+        method: "browsingContext.locateNodes",
+        params: browsingContext.LocateNodesParameters
+      )
+
+      browsingContext.LocateNodesParameters = {
+         context: browsingContext.BrowsingContext,
+         locators: [ + browsingContext.Locator ],
+         ? startNode: script.SharedReference,
+         ? waitTimeout: js-uint
```

Comment:
do we want timeouts to be part of the protocol definitions? this kind of timeouts looks like it should be a client side timeout since the locator is not abortable.

Reply:
@OrKoN I'm not entirely sure what you mean by "not abortable;" the algorithm for element finding does abort upon timing out. Having a timeout for this command solves a number of issues, including being able to wait for an element's presence without a duplicating `waitForElement` command.

I mean, there's another design entirely. where we treat finding elements like network interception, firing off a find command, and having an event tell us when the element is found, if we _really_ want to be edgy and async, but that seems really challenging to get right, and not ergonomic to what developers (especially non-JavaScript developers) expect.

Reply:
I meant that if you, for example, start running a JS locator (applicable to many other locators but maybe others are less likely to timeout), there is no way to abort it even if timeout happens. So what would implementations do is to just throw an ignoring the locator results. The same can be achieved on the client but stopping to await the response message from the protocol. For the "waitForElement" use case it would be indeed useful, but we also need to define a bunch of other parameters such as how many elements to wait for including, potentially, waiting for elements to disappear (i.e., having zero elements matching the selector).

Reply:
I think we probably need to re-introduce timeouts from the Classic spec, and then allow those to be used. A per-request timeout, per-command is going to get pretty ugly, pretty quickly. I'd opt to leave `waitTimeout` out, and then implement it later using the global timeouts.

Reply:
I've removed the JavaScript-based locator for now. Implementation of the general case of #447 would be a more appropriate way to handle this.